lib/puppet/transaction/additional_resource_generator.rb

```diff
@@ -170,8 +170,11 @@ def add_generated_directed_dependency(parent, child, label=nil)
     }
 
     if not edge_exists
+      parameters = target.original_parameters
+      resource = ::Puppet::Resource.new(target.class, target.title, :parameters => parameters)
```

Comment:
I was having issues with namespacing here, and being totally explicit seemed the best way to move forward